File: EMS-project/employ_management_system/src/main/java/com/employ/management/service/EmployeeService.java
package com.employ.management.service;

import java.util.List;

import com.employ.management.entity.Employee;

public interface EmployeeService {

	public Employee saveEmployee(Employee employee);
	
	public List<Employee> getAllEmployees();
	
	public Employee getEmployeeById(Long id);
	
	public Employee updateEmployee(Employee employee);
	
	public void deleteEmployee(Employee employee);
	
	List<Employee> searchEmployees(String keyword);
}

File: EMS-project/ems-frontend/src/services/EmployeeService.js
import axios from "axios";

const REST_API_BASE_URL = 'http://localhost:9090/api/employees';

export const employeesList = () => {
    return axios.get(REST_API_BASE_URL);
}

export const createEmployee = (employee) =>
  axios.post(REST_API_BASE_URL, employee);

export const getEmployeeById = (id) =>
  axios.get(`${REST_API_BASE_URL}/${id}`);

export const updateEmployee = (id, employee) =>
  axios.put(`${REST_API_BASE_URL}/${id}`, employee);

export const deleteEmployee = (id) =>
  axios.delete(`${REST_API_BASE_URL}/${id}`);

export const searchEmployees = (keyword) => {
  return axios.get(
    `http://localhost:9090/api/employees/search?keyword=${keyword}`
  );
};


